### PR TITLE
Create the /etc/mtab file if does not exists

### PR DESCRIPTION
**Pull request by rhatdan**
We should create the /etc/mtab->/proc/mountinfo link
so that mount command will work within the container.

Docker does this by default.

Fixes: https://github.com/containers/podman/issues/10263

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
All kinds of red unhappy tests @rhatdan

**Comment by mheon** [comment]
Can we do this with a bind mount of the link? As is, we’re creating a file
that will show up in diff, be committed with user changes, etc

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
@mheon We already do this, with /etc/resolv.conf and /etc/hosts, if they do not exist in the base image.  We could create an empty file for the inode and mount over it, but I am not sure this is better then setting up the image with the symbolic link, which BTW almost all images have by default, which is why we never caught this until now.  We can talk at standup about potentially special handling the additional mount inodes we add to images.

**Comment by mheon** [comment]
We bind-mount hosts, resolv, and most others (with a few notable
exceptions) - it seems more sane than creating files in the container’s own
RW layer unless there is a good reason not to

File: libpod/container_internal.go
Comment (giuseppe) [inline libpod/container_internal.go:1540]: could this be simplified as?

```
if err = os.Symlink("/proc/mounts", mtab); err != nil && !os.IsExist(err) {
	return "", err
}
```

**Review by giuseppe — APPROVED** [review]
just a nit that could be addressed later,

LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
but I'll let @mheon hit the merge button based on his last comment.

**Pull request merged**
